src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -10,45 +10,397 @@
-    static_cast<int>(val3), \
-    static_cast<int>(val2), \
-    static_cast<int>(val1), \
-    static_cast<int>(val0)  \
-  }
+    SIMD_ALIGN_VAR(static const int, _xmm_const_##name[4], 16) = { \
+        static_cast<int>(val3), \
+        static_cast<int>(val2), \
+        static_cast<int>(val1), \
+        static_cast<int>(val0)  \
+    }
 
 #define SIMD_CONST_PS(name, val0, val1, val2, val3) \
-  SIMD_ALIGN_VAR(static const float, _xmm_const_##name[4], 16) = { \
-    static_cast<float>(val3), \
-    static_cast<float>(val2), \
-    static_cast<float>(val1), \
-    static_cast<float>(val0)  \
-  }
+    SIMD_ALIGN_VAR(static const float, _xmm_const_##name[4], 16) = { \
+        static_cast<float>(val3), \
+        static_cast<float>(val2), \
+        static_cast<float>(val1), \
+        static_cast<float>(val0)  \
+    }
 
 #define SIMD_GET_PS(name) (*(const __m128  *)_xmm_const_##name)
 
+#define RPP_LOAD48_U8PKD3_TO_F32PLN3(srcPtrTemp) \
+{ \
+    px0 = _mm_loadu_si128((__m128i *)srcPtrTemp);           /* load [R01|G01|B01|R02|G02|B02|R03|G03|B03|R04|G04|B04|R05|G05|B05|R06] - Need RGB 01-04 */ \
+    px1 = _mm_loadu_si128((__m128i *)(srcPtrTemp + 12));    /* load [R05|G05|B05|R06|G06|B06|R07|G07|B07|R08|G08|B08|R09|G09|B09|R10] - Need RGB 05-08 */ \
+    px2 = _mm_loadu_si128((__m128i *)(srcPtrTemp + 24));    /* load [R09|G09|B09|R10|G10|B10|R11|G11|B11|R12|G12|B12|R13|G13|B13|R14] - Need RGB 09-12 \ */ \
+    px3 = _mm_loadu_si128((__m128i *)(srcPtrTemp + 36));    /* load [R13|G13|B13|R14|G14|B14|R15|G15|B15|R16|G16|B16|R17|G17|B17|R18] - Need RGB 13-16 \ */ \
+    px0 = _mm_shuffle_epi8(px0, mask);    /* shuffle to get [R01|R02|R03|R04|G01|G02|G03|G04 || B01|B02|B03|B04|R05|G05|B05|R06] - Need R01-04, G01-04, B01-04 */ \
+    px1 = _mm_shuffle_epi8(px1, mask);    /* shuffle to get [R05|R06|R07|R08|G05|G06|G07|G08 || B05|B06|B07|B08|R09|G09|B09|R10] - Need R05-08, G05-08, B05-08 */ \
+    px2 = _mm_shuffle_epi8(px2, mask);    /* shuffle to get [R09|R10|R11|R12|G09|G10|G11|G12 || B09|B10|B11|B12|R13|G13|B13|R14] - Need R09-12, G09-12, B09-12 */ \
+    px3 = _mm_shuffle_epi8(px3, mask);    /* shuffle to get [R13|R14|R15|R16|G13|G14|G15|G16 || B13|B14|B15|B16|R17|G17|B17|R18] - Need R13-16, G13-16, B13-16 */ \
+    px4 = _mm_unpackhi_epi8(px0, zero);    /* unpack 8 hi-pixels of px0 */ \
+    px5 = _mm_unpackhi_epi8(px1, zero);    /* unpack 8 hi-pixels of px1 */ \
+    px6 = _mm_unpackhi_epi8(px2, zero);    /* unpack 8 hi-pixels of px2 */ \
+    px7 = _mm_unpackhi_epi8(px3, zero);    /* unpack 8 hi-pixels of px3 */ \
+    px0 = _mm_unpacklo_epi8(px0, zero);    /* unpack 8 lo-pixels of px0 */ \
+    px1 = _mm_unpacklo_epi8(px1, zero);    /* unpack 8 lo-pixels of px1 */ \
+    px2 = _mm_unpacklo_epi8(px2, zero);    /* unpack 8 lo-pixels of px2 */ \
+    px3 = _mm_unpacklo_epi8(px3, zero);    /* unpack 8 lo-pixels of px3 */ \
+    p0R = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px0, zero));    /* unpack 4 lo-pixels of px0 - Contains R01-04 */ \
```

Comment:
probably you can avoid unpacking by modifying the mask for _mm_shuffle_epi8.

Reply:
We do need to unpack to float to perform the computations though